CRM/Contribute/BAO/Contribution.php

```diff
@@ -5671,4 +5675,25 @@ public static function replaceContributionTokens(
     return $contributionDetails;
   }
 
+  /**
+   * Generate and store invoice_number for contribution.
+   *
+   * @param int $contributionID
+   *
+   * @return string
+   */
+  public static function storeInvoiceNumber($contributionID) {
+    $invoiceNumber = NULL;
+    if ($invoicePrefix = self::checkContributeSettings('invoice_prefix')) {
```

Comment:
Invoice number should be set if Enable Tax and Invoicing is enabled and Invoice Prefix is not empty.